chronicle/release/releasers/github/gh_pull_request.go

```diff
@@ -317,14 +322,28 @@ func fetchMergedPRs(user, repo string) ([]ghPullRequest, error) {
 		}
 
 		// var limit rateLimit
-		for {
+		var (
+			process   bool
+			terminate = false
+		)
+
+		for !terminate {
+			log.WithFields("user", user, "repo", repo, "page", pages).Trace("fetching merged PRs from github.com")
+
 			err := client.Query(context.Background(), &query, variables)
 			if err != nil {
 				return nil, err
 			}
+
 			// limit = query.RateLimit
 
 			for _, prEdge := range query.Repository.PullRequests.Edges {
+				saw++
+				process, terminate = checkSearchTermination(since, &prEdge.Node.UpdatedAt, &prEdge.Node.MergedAt)
```

Comment:
I'm trying to understand the logic here. What I think is happening:

1. We've asked GH for relevant PRs, with more recently updated PRs first.
2. For each PR, we check two things: Should we process the current PR (`process`) and should we break off running additional queries (`terminate`).
3. If an PR was _merged_ before the last release, we don't process it (because it was presumably part of the last release).
4. If a PR was last _updated_ before the last release, we break off searching, because we're ordering GH events by last update, and we got to one before the ones we care about.

Is that right?

Reply:
yup yup -- right on. The go case is that a PR isn't updated after it's merged, so `merge-timestamp == update-timestamp`, but it is possible to update a PR that may or may not be part of this release after it's been merged (e.g. adding a label to the issue, adding a comment, etc)... we want to curtail the search to save on time but not include PRs that were merged before the last release.